okhttp-urlconnection/src/main/java/okhttp3/internal/huc/HttpURLConnectionImpl.java

```diff
@@ -651,4 +563,72 @@ private void setProtocols(String protocolsString, boolean append) {
     this.fixedContentLength = contentLength;
     super.fixedContentLength = (int) Math.min(contentLength, Integer.MAX_VALUE);
   }
+
+  @Override public synchronized void onFailure(Call call, IOException e) {
+    this.callFailure = e;
+    notifyAll();
```

Comment:
Should we worry about application code potentially synchronizing and waiting on this instance? The latter seems unlikely but the former might deadlock something?